yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -976,6 +976,11 @@ private[spark] class Client(
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
           case NonFatal(e) =>
+            if (e.isInstanceOf[InterruptedException]
```

Comment:
then the code segments will be seperated into two parts. i am not sure it's better.